### PR TITLE
SPLICE-1023 add more info in the message for data import error(2.5)

### PR DESCRIPTION
**Pull request by yxia92**

### CODE REVIEW

File: splice_machine/src/main/java/com/splicemachine/derby/stream/function/AbstractFileFunction.java [hunk 1]
```diff
@@ -124,20 +123,34 @@ private void writeNullableUTF(ObjectOutput out, String source) throws IOExceptio
...
+        int numofColumnsinTable = 0;
+        int numofColumnsinFile = 0;
+        boolean columnnumbermistmatch = false;
+
         operationContext.recordRead();
         try {
             ExecRow returnRow = execRow.getClone();
-            if(values==null)
+            if (values == null) {
+                columnnumbermistmatch = true;
```
Comment (snowspeeder) [inline splice_machine/src/main/java/com/splicemachine/derby/stream/function/AbstractFileFunction.java]: why assign to value to local variable before throwing exception?

File: splice_machine/src/main/java/com/splicemachine/derby/stream/function/AbstractFileFunction.java [hunk 2]
```diff
@@ -124,20 +123,34 @@ private void writeNullableUTF(ObjectOutput out, String source) throws IOExceptio
...
                 throw StandardException.newException(SQLState.COLUMN_NUMBER_MISMATCH, execRow.nColumns(), 0);
+            }
+            else if (values.size() < returnRow.nColumns()) {
+                columnnumbermistmatch = true;
```
Comment (snowspeeder) [inline splice_machine/src/main/java/com/splicemachine/derby/stream/function/AbstractFileFunction.java]: same here

File: splice_machine/src/main/java/com/splicemachine/derby/stream/function/AbstractFileFunction.java (hunk 1 above)
Comment (yxia92) [inline splice_machine/src/main/java/com/splicemachine/derby/stream/function/AbstractFileFunction.java]: The variable columnnumbermismatch is used later in processing the exception in the catch part.

Comment (snowspeeder) [inline splice_machine/src/main/java/com/splicemachine/derby/stream/function/AbstractFileFunction.java]: I see, exception is being caught in the same function body later

**Pull request merged**